### PR TITLE
sci-electronics/librepcb: version up to 1.1.0

### PR DESCRIPTION
**Pull request by ktrace**
+ add qt6 build variant

### CODE REVIEW

File: sci-electronics/librepcb/librepcb-1.1.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+DESCRIPTION="Free EDA software to develop printed circuit boards"
+HOMEPAGE="https://librepcb.org/"
+SRC_URI="https://download.librepcb.org/releases/${PV}/${P}-source.zip"
+
+LICENSE="GPL-3+"
+SLOT="0"
+KEYWORDS="~amd64"
+
+IUSE="opencascade +qt5 qt6"
+REQUIRED_USE="|| ( qt5 qt6 )"
```
Comment (parona-source) [inline sci-electronics/librepcb/librepcb-1.1.0.ebuild:17]: https://wiki.gentoo.org/wiki/Project:Qt/Policies#Requires_one_of_two_Qt_versions

qt5 and qt6 cant be on at the same time so the qt5 use flag does nothing other than pull in the dependencies

Comment (ktrace) [reply sci-electronics/librepcb/librepcb-1.1.0.ebuild:17]: But what I mean is https://wiki.gentoo.org/wiki/Project:Qt/Policies#Requires_at_least_one_Qt_version

Comment (parona-source) [reply sci-electronics/librepcb/librepcb-1.1.0.ebuild]: But this ebuild doesn't do that. If you enable qt6 you dont get qt5. If you wanted to that you would have to add it with multibuild, but even so I dont see why you would do that.

> -DQT_MAJOR_VERSION=$(usex qt6 6 5)

File: sci-electronics/librepcb/librepcb-1.1.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+			dev-qt/qtnetwork:5[ssl]
+			dev-qt/qtopengl:5
+			dev-qt/qtprintsupport:5
+			dev-qt/qtquickcontrols2:5
+			dev-qt/qtsql:5
+			dev-qt/qtsvg:5
+			dev-qt/qtwidgets:5
+			dev-qt/qtxml:5
+	)
+	qt6? (  dev-libs/quazip:=[qt6(+)]
```
Comment (parona-source) [inline sci-electronics/librepcb/librepcb-1.1.0.ebuild:40]:
```suggestion
	qt6? (  dev-libs/quazip:=[qt6(-)]
```

quazip-1.2 has qt5 support but not qt6 support. Meaning the qt5 assumption is correct, but not the qt6 assumption.

File: sci-electronics/librepcb/librepcb-1.1.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+			dev-qt/qtbase[concurrent,dbus,gui,network,opengl,ssl,sql,sqlite,test,widgets,xml] )
+	opencascade? ( sci-libs/opencascade:= )
+	sys-libs/zlib
+	virtual/opengl"
+
+DEPEND="${RDEPEND}
+	!qt6? ( dev-qt/qttest:5 )
+	qt6? ( dev-qt/qtbase:6[test] )"
```
Comment (parona-source) [inline sci-electronics/librepcb/librepcb-1.1.0.ebuild:48]:
```suggestion
```

qtbase:6 includes the modules needed for testing without the test use flag.

File: sci-electronics/librepcb/librepcb-1.1.0.ebuild (hunk 1 above)
Comment (ktrace) [reply sci-electronics/librepcb/librepcb-1.1.0.ebuild]: After chat on #gentoo-qt I realize you're right.

File: sci-electronics/librepcb/librepcb-1.1.0.ebuild (hunk 2 above)
Comment (ktrace) [reply sci-electronics/librepcb/librepcb-1.1.0.ebuild]: thank you, fixed

File: sci-electronics/librepcb/librepcb-1.1.0.ebuild (hunk 3 above)
Comment (ktrace) [reply sci-electronics/librepcb/librepcb-1.1.0.ebuild]: thank you, fixed

**Comment by AndrewAmmerlaan** [comment]
Thanks :+1:

**Pull request closed without merging**